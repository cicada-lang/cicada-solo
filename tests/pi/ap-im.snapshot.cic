id: (given A: Type, x: A) -> A =
  (x) {
    x
  }

id(1)
id(given Nat, 1)

id("a")
id(given String, "a")
