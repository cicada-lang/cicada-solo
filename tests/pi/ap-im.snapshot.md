# implicit argument insertion when applying implicit function

``` cicada
id(given A: Type, x: A): A {
  x
}

id(1)
id(given Nat, 1)

id("a")
id(given String, "a")
```

# multiple implicit arguments

``` cicada
k(
  given A: Type, x: A,
  given B: Type, y: B,
): A {
  x
}

k(100, 101)
k(given Nat, 100, given Nat, 101)
```
