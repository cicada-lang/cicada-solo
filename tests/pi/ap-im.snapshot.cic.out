1: Nat 
1: Nat 
"a": String 
"a": String 
1: Nat 
1: Nat 
"a": String 
"a": String 

